src/merge.ts

```diff
@@ -0,0 +1,135 @@
+    // Fetch PR data
+    let pull_request = (await octokit.rest.pulls.get({
+        owner: repository.owner.login,
+        repo: repository.name,
+        pull_number: pull_number
+    })).data;
+    const title = pull_request.title;
+
+    // Modify EIP data when needed
+    let newFiles = [];
+    for (let file of files) {
+        file = { ...file };
+        if (file.filename.endsWith('.md')) {
+            // Parse file
+            const fileContent = file.contents as string;
+            const fileData = fm(fileContent);
+            const frontmatter = fileData.attributes as FrontMatter;
+
+            // Check if EIP number needs setting
+            if (!frontmatter.eip) {
+                let eip = await generateEIPNumber(octokit, repository);
+
+                frontmatter.eip = `${eip}`;
+                file.filename = `EIPS/eip-${eip}.md`;
+            }
+
+            // Check if status needs setting
+            if (!frontmatter.status) {
+                frontmatter.status = "Draft";
+            }
+
+            // Check if last call deadline needs setting
+            if (frontmatter.status == "Last Call" && !frontmatter["last-call-deadline"]) {
+                let fourteenDays = new Date(Date.now() + 12096e5);
+                frontmatter["last-call-deadline"] = `${fourteenDays.getUTCFullYear()}-${fourteenDays.getUTCMonth()}-${fourteenDays.getUTCDate()}`;
+            }
+
+            // Now, regenerate markdown from front matter
+            file.contents = `---\n${yaml.dump(frontmatter)}\n---${fileData.body}`;
+            
+            // Push
+            newFiles.push(file);
+        }
+    }
+
+    // Push changes
+    for (let i = 0; i < files.length; i++) {
+        let oldFile = files[i];
+        let newFile = newFiles[i];
+
+        if (oldFile.status == "removed") {
+            continue;
+        }
+
+        if (oldFile.filename !== newFile.filename) {
+            // Delete old
+            await octokit.rest.repos.deleteFile({
+                owner: pull_request.head.repo?.owner?.login as string,
+                repo: pull_request.head.repo?.name as string,
+                path: newFile.filename,
+                sha: oldFile.sha as string,
+                message: `Delete ${oldFile.filename}`,
+                committer: {
+                    name: "eth-bot",
+                    email: localConfig.commitEmail
+                },
+            });
+        }
+        await octokit.rest.repos.createOrUpdateFileContents({
+            owner: pull_request.head.repo?.owner?.login as string,
+            repo: pull_request.head.repo?.name as string,
+            path: newFile.filename,
+            sha: oldFile.sha as string,
+            message: `Update ${newFile.filename}`,
+            committer: {
+                name: "eth-bot",
+                email: localConfig.commitEmail
+            },
+            content: Buffer.from(newFile.contents as string).toString('base64') as string
+        });
```

Comment:
This creates several commits, right? We need it to be atomic.

Reply:
Well, we can't have that with GitHub's current API. It will only create one commit on the EIPs repository, though.